### PR TITLE
Cassandra changes

### PR DESCRIPTION
**Pull request by toxeh**

### CODE REVIEW

**Comment by pnowojski** [comment]
% those cosmetic comments it looks good to me

**Comment by pnowojski** [comment]
One more thing @toxeh - travis has failed with some check style issues, which need to be resolved

File: presto-cassandra/pom.xml [hunk 1]
```diff
@@ -13,12 +13,19 @@
...
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <cassandra.version>2.1.0</cassandra.version>
+        <cassandra.version>2.1.3</cassandra.version>
         <datastax.version>2.1.5</datastax.version>
     </properties>
 
     <dependencies>
         <dependency>
+            <groupId>org.slf4j</groupId>
```
Comment (martint) [inline presto-cassandra/pom.xml]: Please, don't use slf4j for logging. Either use java.util.logging or airlift's lightweight wrapper (https://github.com/airlift/airlift/blob/master/log/src/main/java/io/airlift/log/Logger.java)

**Comment by pnowojski** [comment]
Thanks, lgtm

**Comment by martint** [comment]
@toxeh can you squash your commits (into independent commits if appropriate) and add a description about what each one does to the commit message?

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTokenSplitManager.java [hunk 2]
```diff
@@ -43,7 +43,7 @@
     private final CassandraThriftClient cassandraThriftClient;
     private final ExecutorService executor;
     private final int splitSize;
-    private final IPartitioner<?> partitioner;
+    private final IPartitioner partitioner;
```
Comment (martint) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTokenSplitManager.java]: IPartitioner is a generic class. Don't use the raw type.

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/LimitedLocalNodeFirstLocalBalancingPolicy.java [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+ * It chooses alive hosts only from the set of the given contact points - because the connection is used to load the data from
+ * the particular split, with a strictly defined list of the allowed hosts, it is pointless to try the other nodes.
+ * The policy tracks which of the allowed hosts are alive, and when a new query plan is requested, it returns those hosts
+ * in the following order:
+ * <ul>
+ * <li>the local node</li>
+ * <li>the collection of the remaining hosts (which is shuffled on each request)</li>
+ * </ul>
+ */
+class LimitedLocalNodeFirstLocalBalancingPolicy implements LoadBalancingPolicy
```
Comment (martint) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/LimitedLocalNodeFirstLocalBalancingPolicy.java]: `implements` goes on the next line

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/LimitedLocalNodeFirstLocalBalancingPolicy.java [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    {
+        InetAddress hostAddress = host.getAddress();
+        return hostAddress.isLoopbackAddress() || localAddresses.contains(hostAddress);
+    }
+
+    private static Set<InetAddress> getLocalInetAddresses()
+    {
+        try {
+            return Sets.newHashSet(Iterators.concat(
+                    Iterators.transform(
```
Comment (martint) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/LimitedLocalNodeFirstLocalBalancingPolicy.java]: Use Java 8 streams instead of Guava Iterables

File: presto-cassandra/src/main/java/com/facebook/presto/cassandra/LimitedLocalNodeFirstLocalBalancingPolicy.java [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+    private static boolean isLocalHost(Host host)
+    {
+        InetAddress hostAddress = host.getAddress();
+        return hostAddress.isLoopbackAddress() || localAddresses.contains(hostAddress);
+    }
+
+    private static Set<InetAddress> getLocalInetAddresses()
+    {
+        try {
+            return Sets.newHashSet(Iterators.concat(
```
Comment (martint) [inline presto-cassandra/src/main/java/com/facebook/presto/cassandra/LimitedLocalNodeFirstLocalBalancingPolicy.java]: Just call `new HashSet<>(...)`

File: presto-tests/pom.xml [hunk 6]
```diff
@@ -173,4 +173,15 @@
             <scope>runtime</scope>
         </dependency>
     </dependencies>
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <forkMode>always</forkMode>
```
Comment (martint) [inline presto-tests/pom.xml]: Why is this needed?

**Comment by martint** [comment]
I'm going to close this for now since there hasn't been any activity in a few months. Please, reopen if the the changes are still relevant and once the comments have been addressed.
